### PR TITLE
Add Gaussian Process Regression analysis for EQSANS I(Q) profiles

### PR DESCRIPTION
**Pull request by darshdinger**
## Description of work:

This PR implements Gaussian Process Regression (GPR) analysis for EQSANS I(Q) profiles during autoreduction. GPR provides uncertainty-quantified fits that help visualize trends in noisy scattering data. The analysis automatically generates interactive Plotly plots embedded in HTML reports and high-resolution PNG images.

**What's included:**
- **Core GPR module** (`src/drtsans/gpr.py`): 910-line implementation with RBF kernel, log-space transforms, and iterative background estimation. Algorithm adapted from Changwoo Do (ORNL).
- **Autoreduction integration**: Seamlessly added to `reduce_EQSANS.py` with graceful error handling. Runs by default but can be disabled via `enableGPR: false` configuration.
- **CLI tool**: Command-line interface with tab completion (argcomplete) and interactive TUI (argparse-tui) for standalone analysis.
- **Comprehensive testing**: 34 tests (24 unit + 10 integration) covering normal cases, edge cases, and error conditions. 100% pass rate.
- **Complete documentation**: New user guide (`docs/user/gpr_analysis.rst`) with usage examples, parameter tuning, and troubleshooting. Updated autoreduction docs with GPR output files.

**Key features:**
- Handles single profiles, multiple wedges, and time slices
- Produces PNG (300 DPI) and DAT files alongside reduction outputs
- Never breaks reduction workflow (isolated error handling)
- Works in headless environments (matplotlib Agg backend)
- Follows all drtsans coding standards (type hints, Mantid logger, ruff compliant)

**Check all that apply:**
- [x] updated documentation and checked that it looks correct in the [pull request preview](https://docs.readthedocs.com/platform/stable/pull-requests.html)
- [x] Source added/refactored
- [x] Added unit tests
- [x] Added integration tests
- [x] Included a manual test for the reviewer
- [ ] Verified that tests requiring the /SNS and /HFIR filesystems pass without fail

**References:**
- Links to IBM EWM items: [EWM 13811](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20(Change%20Management)#action=com.ibm.team.workitem.viewWorkItem&id=13811)
- Links to related issues or pull requests: [Add if applicable]
- User Story: Gaussian Process Regression plot for I(Q) during EQSANS autoreduction

### Test 1: Standalone GPR CLI
```bash
pixi shell
cd tests/integration/drtsans/
python -m drtsans.gpr --help  # Should show all options including --tui
exit
```

### Test 2: Run GPR tests
```bash
pixi shell
pytest tests/unit/drtsans/test_gpr.py -v
pytest tests/integration/drtsans/test_gpr.py -v
exit
```
Expected: All 34 tests pass (24 unit + 10 integration)

### Test 3: GPR on sample data file
```bash
pixi shell
# Create a simple test I(Q) file
cat > test_iq.dat << 'EOF'
# Q (1/A)	I (1/cm)	I_err (1/cm)	dQ (1/A)
0.010000	1.000e+06	1.000e+05	0.0005
0.020000	2.500e+05	2.500e+04	0.0010
0.050000	4.000e+04	4.000e+03	0.0025
0.100000	1.000e+04	1.000e+03	0.0050
0.200000	2.500e+03	2.500e+02	0.0100
0.500000	4.000e+02	4.000e+01	0.0250
EOF

# Run GPR analysis
python -m drtsans.gpr test_iq.dat --output-dir /tmp

# Check outputs (should see PNG plot and DAT file)
ls -lh /tmp/test_iq_gpr.*
exit
```

## Check list for the reviewer
- [ ] best software practices
    + [ ] clearly named variables (better to be verbose in variable names)
    + [ ] code comments explaining the intent of code blocks
- [ ] All the tests are passing
- [ ] The documentation is up to date and looks correct in the [pull request preview](https://docs.readthedocs.com/platform/stable/pull-requests.html)
- [ ] code comments added when explaining intent

### CODE REVIEW

File: scripts/autoreduction/reduce_EQSANS.py [hunk 1]
```diff
@@ -308,6 +315,104 @@ def reduce_non_sample(events: EventWorkspace):
...
+
+            else:
+                # Single profile (scalar or annular binning)
+                base_name = f"{output_filename}{slice_suffix}"
+                logger.info(f"Running GPR for profile set {i}")
+
+                try:
+                    # generate_gpr_analysis can handle list or single IQmod
+                    html, png_files, dat_files = generate_gpr_analysis(
+                        profiles[0] if len(profiles) == 1 else profiles, output_dir, base_name
```
Comment (jmborr) [inline scripts/autoreduction/reduce_EQSANS.py]: When argument `iqmod_objects` is an instance of `IQmod`, function `generate_gpr_analysis` wraps it into a list so why pass `profiles[0]`?

File: scripts/autoreduction/reduce_EQSANS.py [hunk 2]
```diff
@@ -308,6 +315,104 @@ def reduce_non_sample(events: EventWorkspace):
     return report
 
 
+def generate_gpr_plots(
```
Comment (jmborr) [inline scripts/autoreduction/reduce_EQSANS.py]: move this function to module gpr.py so that we don't load the autoreduction script with this particular extension. 
In the script, one would simply write `from drtsans.extenstions.gpr import generate_gpr_plots` with perhaps a more meaningful name such as `autoreduction_plots`.
As added benefit, you can test this function in the `test_grp.py` files

**Review by jmborr — CHANGES_REQUESTED** [review]

File: scripts/autoreduction/reduce_EQSANS.py (hunk 1 above)
Comment (darshdinger) [reply scripts/autoreduction/reduce_EQSANS.py]: You are right, this is just added complexity. The updated code now just passes 'profiles' directly:
```python
html, png_files, dat_files = generate_gpr_analysis(profiles, output_dir, base_name)
```

File: scripts/autoreduction/reduce_EQSANS.py (hunk 2 above)
Comment (darshdinger) [reply scripts/autoreduction/reduce_EQSANS.py]: Done.

File: extensions/gpr.py [hunk 3]
```diff
@@ -0,0 +1,1072 @@
+"""
```
Comment (jmborr) [inline extensions/gpr.py:1]: Extensions are not supposed to modify the reduction codebase, hence `extensions/` should be placed not under `src/drtsans` but at the top level level of the repo, in the same way as `scripts/` or `notebook/`.  
Unit tests should go under `tests/unit/extensions/`.  
Integration tests should go under `tests/unit/extensions/` and/or other `test/integration/` subdirectories like `test/integration/autoreduction/`

File: scripts/autoreduction/reduce_EQSANS.py [hunk 4]
```diff
@@ -398,6 +405,17 @@ def reduce_sample(events: EventWorkspace, output_dir: str, logger: logging.Logge
     logger.info("reduce_sample: creating Plotly output with plotly_reduction_output()")
     report += plotly_reduction_output(output, input_config) + "<hr>\n"
 
+    # Generate GPR analysis plots
+    logger.info("reduce_sample: generating GPR analysis plots")
+    gpr_report = autoreduction_plots(
```
Comment (jmborr) [inline scripts/autoreduction/reduce_EQSANS.py]: `autoreduction_plots` won't be available when `GPR_AVAILABLE` is `False`

**Review by jmborr — CHANGES_REQUESTED** [review]

File: scripts/autoreduction/reduce_EQSANS.py (hunk 4 above)
Comment (darshdinger) [reply scripts/autoreduction/reduce_EQSANS.py]: In this next commit if the extensions.gpr module fails to import, the function won't be called and autoreduction will continue normally without GPR plots. Thanks for catching that. :+1:

File: extensions/gpr.py (hunk 3 above)
Comment (darshdinger) [reply extensions/gpr.py:1]: Okay, understood. Hopefully now I have corrected the file structure. Let me know if it is still incorrect. :)

File: pyproject.toml [hunk 5]
```diff
@@ -23,6 +23,7 @@ BugTracker = "https://github.com/neutrons/drtsans/issues"
 
 [project.scripts]
 grasp_cg2 = "drtsans.mono.gpsans.export_to_grasp:main"
+drtsans-gpr = "extensions.gpr:main"
```
Comment (jmborr) [inline pyproject.toml]: Have you verified this works? I mean if you have created the conda package and then installed it into some sandbox environment.
I believe you'll need to add
```
[tool.hatch.build.targets.wheel.force-include]
"extensions" = "drtsans/extensions"
```
to pyproject.toml so that directory `extensions/` will make it in the conda package. Later when you install the conda package in an environment, you should be able to write statements such as
```
from drtsans.extensions import gpr
```
You'll need to add an `__init__.py` file under `extensions/`, and you can simply have the shorcut:
```
[project.scripts]
gpr = "drtsans.extensions.gpr:main"
```

**Review by jmborr — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 5 above)
Comment (darshdinger) [reply pyproject.toml]: I had forgotten to update this with my previous change. Thanks for the suggestion. I will go ahead and implement.

Comment (darshdinger) [reply pyproject.toml]: Verified that CLI works in both modes:
```bash
# Development mode
python -m extensions.gpr --help

# Installed package mode (after building wheel)
python -m drtsans.extension.gpr --help
```

File: scripts/autoreduction/reduce_EQSANS.py [hunk 6]
```diff
@@ -35,11 +35,16 @@
 from drtsans.tof.eqsans.meta_data import is_sample_run
 
 try:
-    from extensions.gpr import autoreduction_plots
+    from drtsans.extensions.gpr import autoreduction_plots
 
     GPR_AVAILABLE = True
 except ImportError:
-    GPR_AVAILABLE = False
+    try:
```
Comment (jmborr) [inline scripts/autoreduction/reduce_EQSANS.py:42]: good catch!

File: tests/integration/extensions/test_gpr_integration.py [hunk 7]
```diff
@@ -10,7 +10,12 @@
 
 # local imports
 from drtsans.dataobjects import IQmod
-from extensions.gpr import generate_gpr_analysis, run_gpr_from_file
+
+try:
+    from drtsans.extensions.gpr import generate_gpr_analysis, run_gpr_from_file
```
Comment (jmborr) [inline tests/integration/extensions/test_gpr_integration.py]: do you envision other mode than Development mode for running the tests? After all, we don't ship the tests in the conda package 🤔

File: tests/unit/extensions/test_gpr_unit.py [hunk 8]
```diff
@@ -8,16 +8,30 @@
...
-    f_loglin_deriv,
-    run_gpr,
-    create_png_plot,
-    plotly_gpr_plot,
-    generate_gpr_analysis,
-    run_gpr_from_file,
-)
+
+try:
+    from drtsans.extensions.gpr import (
```
Comment (jmborr) [inline tests/unit/extensions/test_gpr_unit.py]: same here. Tests are run in Development mode

**Review by jmborr — APPROVED** [review]
👌
